centreon/src/Core/Common/Infrastructure/Repository/SqlMultipleBindTrait.php

```diff
@@ -28,16 +28,17 @@ trait SqlMultipleBindTrait
     /**
      * @param array<int|string, int|string> $list
      * @param string $prefix
+     * @param int $bindType (optional)
      *
      * @return array{0: array<string, mixed>, 1: string}
      */
-    protected function createMultipleBindQuery(array $list, string $prefix): array
+    protected function createMultipleBindQuery(array $list, string $prefix, ?int $bindType = null): array
     {
         $bindValues = [];
         $list = array_values($list); // Reset index to avoid SQL injection
 
         foreach ($list as $index => $id) {
-            $bindValues[$prefix . $index] = $id;
+            $bindValues[$prefix . $index] = ($bindType === null) ? $id : [$id, $bindType];
```

Comment:
You must comment out the new return values when the optional parameter is sent.

Reply:
resolved